src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,44 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo | null;
+  setModalOpen: (val: boolean) => void;
+  resetSelectedTodo: (val: Todo | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  setModalOpen,
+  todo,
+  resetSelectedTodo,
+}) => {
+  const [fetchedUser, setFetchedUser] = useState<User | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (todo) {
+      getUser(todo.userId)
+        .then(setFetchedUser)
+        .catch(err => {
+          setError(`Error fetching user: ${err.message}`);
```

Comment:
Ensure that `err.message` is a valid property. If the error object structure is different, this might lead to undefined messages. Consider logging the error to verify its structure.